cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,248 @@
+
+	Clientset k8s.Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewPodOptions(streams genericiooptions.IOStreams) *PodOptions {
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods [--job job/<job-name>]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Validate())
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addJobFlagVar(cmd, &o.JobArg)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Command, args []string) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printPodTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	if len(args) > 0 {
+		jobArg, err := cmd.Flags().GetString("job")
+		if err != nil {
+			return err
+		}
+		o.JobArg = jobArg
+	}
+
+	return nil
+}
+
+func (o *PodOptions) Validate() error {
+	if !o.validJobFlagOptionProvided() {
+		return errors.New("not a valid --job flag. Please provide a valid job name in format job/job-name")
+	}
+	return nil
+}
+
+func (o *PodOptions) validJobFlagOptionProvided() bool {
+	jobArgSlice := strings.Split(o.JobArg, "/")
+
+	// jobArgSlice should be ["job", "job-name"]
+	if len(jobArgSlice) != 2 {
+		return false
+	}
+
+	// valid only if the kind is a job
+	kind := strings.ToLower(jobArgSlice[0])
+	return strings.Contains(kind, "job")
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	jobName := strings.Split(o.JobArg, "/")[1]
+	controllerUID, err := o.getJobControllerUID(ctx, jobName)
+	if err != nil {
+		return err
+	}
+
+	err = o.listPodsByControllerUID(ctx, controllerUID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getJobControllerUID fetches the controllerUID of the given Job
+func (o *PodOptions) getJobControllerUID(ctx context.Context, jobName string) (string, error) {
+	job, err := o.Clientset.BatchV1().Jobs(o.Namespace).Get(ctx, jobName, metav1.GetOptions{})
```

Comment:
I think we missed another job kinds.

Reply:
@mbobrovskyi -- Do you know what other job kinds we need to support? From the above comment `pytorchjob, rayjob, Job, CronJob` anything else?

Reply:
https://kueue.sigs.k8s.io/docs/reference/kueue-config.v1beta1/#Integrations

Reply:
But, we need to find solution how to get it automatically just by Kind, api-group and name to be sure that if we add new framework it will working fine. The biggest problem that we can't find by [field-selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#list-of-supported-fields). 

Maybe the easiest way is just to filter pods by `ownerReferences`.

```yaml
  ownerReferences:
  - apiVersion: kubeflow.org/v1
    blockOwnerDeletion: true
    controller: true
    kind: PyTorchJob
    name: pytorch-simple
    uid: 23417835-a7b7-4b66-aa11-e94bdf0871b8
  resourceVersion: "1133351"
  uid: eb5d7f97-a4bf-4f4c-b7f1-aab361104f17
```

Reply:
Yeah that's what I'm looking into. `OwnerReferences` can work for different kinds.